### ISSUE TITLE
Assist: "add derive `Clone`/`Default`" if no method `clone`/`default` found in current scope

### ISSUE THREAD

**Issue by pickx** [rust-lang/rust-analyzer#15285]
```rust
struct A;

fn main() {
    let a = A;
    a.$0clone();
}
```
or
```rust
struct B;

fn main() {
    B::$0default();
}
```

this currently suggests "generate `clone` method"/"generate `default` function" but I think in most cases the user would prefer to add `#[derive(Clone)]`/`#[derive(Default)]` to the struct/enum (unless the struct/enum cannot derive `Clone`/`Default`).

if this is okay, I would like to try implementing this.

**Comment by Veykril** [comment]
Ideally this should be a diagnostic fix, but the diagnostic is still marked experimental as it has some false positives. So starting this out as an assist seems fine since usually that code can be transitioned to the diagnostic fixes easily later on.

Going with the derive first seems fine as well, since we have an assist to turn a derive into a concrete impl.

**Comment by flodiebold** [comment]
We mark the type mismatch as non-experimental if it has fixes, we could just do the same for unresolved methods.
